### PR TITLE
chore: improve test & misc. setup

### PR DESCRIPTION
**Pull request by 0xCryptoSheik**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1 +1,14 @@
-module.exports = {};
+module.exports = {
+  testRegex: '(/__tests__/.*|(\\.|/)spec)\\.ts$',
```
Comment (0xCryptoSheik) [inline jest.config.js:2]: only `__tests__/**/*.spec.ts` valid test files

File: jest.config.js [hunk 2]
```diff
@@ -1 +1,14 @@
...
+  transform: {
+    '\\.ts$': [
```
Comment (0xCryptoSheik) [inline jest.config.js:4]: transform ts files with babel on the fly when testing w/ Jest

File: package.json [hunk 3]
```diff
@@ -38,23 +38,24 @@
...
     "@types/big.js": "^6.1.3",
-    "@types/jest": "^27.4.1",
-    "@types/node": "^17.0.24",
-    "@typescript-eslint/eslint-plugin": "^5.19.0",
-    "@typescript-eslint/parser": "^5.19.0",
-    "eslint": "^8.13.0",
+    "@types/jest": "^27.5.0",
+    "@typescript-eslint/eslint-plugin": "^5.22.0",
+    "@typescript-eslint/parser": "^5.22.0",
+    "eslint": "^8.15.0",
```
Comment (0xCryptoSheik) [inline package.json:54]: Just minor updates of dev dependencies

File: package.json [hunk 4]
```diff
@@ -38,23 +38,24 @@
     "validate": "npm run typecheck && npm run test && npm run lint && npm run format-check"
   },
   "dependencies": {
-    "@blockworks-foundation/mango-client": "3.4.4",
+    "@blockworks-foundation/mango-client": "3.4.5",
```
Comment (0xCryptoSheik) [inline package.json:41]: Assuming mango respects semver, but no changelog ..

File: tsconfig.json [hunk 5]
```diff
@@ -6,6 +6,7 @@
     "paths": {
       "*": ["./typings/*"]
     },
+    "types": ["big.js", "jest"],
```
Comment (0xCryptoSheik) [inline tsconfig.json:9]: slight typechecking performance potential optimization + explicitness

**Review by acamill — APPROVED** [review]

**Pull request merged**